stream/src/main/scala/org/apache/pekko/stream/MapAsyncPartitioned.scala

```diff
@@ -128,27 +55,32 @@ private[stream] object MapAsyncPartitioned {
 
     override def apply(t: Try[Out]): Unit = {
       setOut(t)
-      callback.invoke(this)
+      cb.invoke(this)
     }
+
+    override def toString = s"Holder($in, $out)"
   }
 }
 
-private[stream] class MapAsyncPartitioned[In, Out, Partition](
-    orderedOutput: Boolean,
+/**
+ * Internal API
+ */
+@InternalApi
+private[stream] final class MapAsyncPartitioned[In, Out, Partition](
     parallelism: Int,
-    extractPartition: In => Partition,
+    orderedOutput: Boolean,
+    partitioner: In => Partition,
     f: (In, Partition) => Future[Out]) extends GraphStage[FlowShape[In, Out]] {
+  require(parallelism >= 1, "parallelism must be at least 1")
+  require(partitioner != null, "partitioner function should not be null")
```

Comment:
Do we need to do the null checks? They have a small overhead.

Reply:
This should be fine, because it only occur at dsl build time not run time, otherwise it will throw NPE when the stage is running

Reply:
Indeed this is correct, they will only occur once when DSL is building so its fine.